standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -304,8 +304,8 @@ static ThriftHiveMetastore.Iface callEmbeddedMetastore(Configuration conf) throw
     try {
       Class<?> clazz = Class.forName(HIVE_METASTORE_CLASS);
       //noinspection JavaReflectionMemberAccess
-      Method method = clazz.getDeclaredMethod(HIVE_METASTORE_CREATE_HANDLER_METHOD,
-          Configuration.class);
+      String methodName = MetastoreConf.getVar(conf, MetastoreConf.ConfVars.HMS_HANDLER_CREATE);
+      Method method = clazz.getDeclaredMethod(methodName,Configuration.class);
```

Comment:
the `newHMSHandler` looks enough to create an embedded client, is there any reason to ask for a new method?

Reply:
It allows specifying the alternate method 'newHMSRetryingLocalHandler' which wraps the normal handler into a retrying RetryingHMSHandler which is missing from the 'raw handler' (if I'm not mistaken).

Reply:
The `newHMSHandler` serves the same purpose, it will wrap the raw handler into a retrying handler